File: 1984-maximum-distance-between-a-pair-of-values/maximum-distance-between-a-pair-of-values.js
var maxDistance = function (nums1, nums2) {
    const m = nums1.length
    const n = nums2.length
    let i = 0, j = 0
    let result = 0
    while (i < m && j < n) {
        if (nums1[i] > nums2[j]) {
            i++
        } else {
            result = Math.max(result, j - i)
            j++
        }
    }
    return result
};

File: 1984-maximum-distance-between-a-pair-of-values/maximum-distance-between-a-pair-of-values.cpp
class Solution {
public:
    int maxDistance(vector<int>& nums1, vector<int>& nums2) {
        int m = nums1.size();
        int n = nums2.size();
        int i = 0, j = 0;
        int result = 0;
        while (i < m && j < n) {
            if (nums1[i] > nums2[j]) {
                i++;
            } else {
                result = max(result, j - i);
                j++;
            }
        }
        return result;
    }
};